### PR TITLE
Add report time feature

### PR DESCRIPTION
**Pull request by alindesign**
**What kind of change does this PR introduce?**
Feature

**Did you add tests for your changes?**
Yes, i have modified

**Summary**
Added option to report time for webpack messages eg: `[12:00:00] webpack: Compiled Successfully.`

**Does this PR introduce a breaking change?**
No

**Other information**
When i run test on my local, Server tests suite have some errors, but i don't have changed this tests.

### CODE REVIEW

File: test/Reporter.test.js [hunk 1]
```diff
@@ -56,8 +57,9 @@ describe("Reporter", function() {
 
 			plugins.done(simpleStats);
 			setTimeout(function() {
+
 				should.strictEqual(console.log.callCount, 2);
-				should.strictEqual(console.log.calledWith("webpack: Compiled successfully."), true);
+				should.strictEqual(console.log.calledWith(getTime() + "webpack: Compiled successfully."), true);
```
Comment (shellscape) [inline test/Reporter.test.js]: these tests shouldn't be modified, they should be left as-is and new tests created when the option is set.

also, there shouldn't be a need for a separate `GetTime` module. you can use a regular expression to check that the timestamp is being prepended to the message.

File: lib/Shared.js [hunk 2]
```diff
@@ -31,10 +34,15 @@ module.exports = function Shared(context) {
 			context.options = options;
 		},
 		defaultReporter: function(reporterOptions) {
+			var time = "''";
```
Comment (shellscape) [inline lib/Shared.js]: I don't understand the empty quotes within this string here

File: lib/GetTime.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+var twoDigit = function(digit) {
```
Comment (shellscape) [inline lib/GetTime.js]: a separate module really isn't needed here. use https://www.npmjs.com/package/time-stamp

File: lib/Shared.js [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 var MemoryFileSystem = require("memory-fs");
+var getTime = require("./GetTime");
+
 var HASH_REGEXP = /[0-9a-f]{10,}/;
 
 module.exports = function Shared(context) {
 	var share = {
 		setOptions: function(options) {
 			if(!options) options = {};
+			if(typeof options.reportTime === "undefined") options.reportTime = true;
```
Comment (shellscape) [inline lib/Shared.js]: this should be `false` by default

File: lib/Shared.js [hunk 5]
```diff
@@ -31,10 +34,15 @@ module.exports = function Shared(context) {
 			context.options = options;
 		},
 		defaultReporter: function(reporterOptions) {
+			var time = "''";
 			var state = reporterOptions.state;
 			var stats = reporterOptions.stats;
 			var options = reporterOptions.options;
 
+			if(!!options.reportTime) {
+				time = getTime();
```
Comment (shellscape) [inline lib/Shared.js]:
```js
time = '[' + timestamp('HH:mm:ss'))+']';
```

File: lib/Shared.js [hunk 6]
```diff
@@ -43,17 +51,20 @@ module.exports = function Shared(context) {
 				if(displayStats) {
 					options.log(stats.toString(options.stats));
 				}
+
```
Comment (shellscape) [inline lib/Shared.js]: please don't add your own whitespace

**Comment by shellscape** [comment]
@alindesign ping

**Comment by alindesign** [comment]
@shellscape i have made your changes, is ok now? sorry for late, i forget to change 🤦‍♂️

File: lib/Shared.js (hunk 6 above)
Comment (shellscape) [inline lib/Shared.js]: @alindesign please remove this extra whitespace

File: lib/Shared.js [hunk 7]
```diff
@@ -43,17 +50,20 @@ module.exports = function Shared(context) {
...
 				if(!options.noInfo && !options.quiet) {
 					var msg = "Compiled successfully.";
+
 					if(stats.hasErrors()) {
 						msg = "Failed to compile.";
 					} else if(stats.hasWarnings()) {
 						msg = "Compiled with warnings.";
 					}
-					options.log("webpack: " + msg);
+
```
Comment (shellscape) [inline lib/Shared.js]: please remove this extra whitespace

File: lib/Shared.js [hunk 8]
```diff
@@ -43,17 +50,20 @@ module.exports = function Shared(context) {
 				if(displayStats) {
 					options.log(stats.toString(options.stats));
 				}
+
 				if(!options.noInfo && !options.quiet) {
 					var msg = "Compiled successfully.";
+
```
Comment (shellscape) [inline lib/Shared.js]: please remove this extra whitespace

**Comment by shellscape** [comment]
@alindesign looks good, but you have some whitespace to remove yet, and there are conflicts that must be resolved. Please rebase to master and resolve those conflicts.

**Comment by alindesign** [comment]
@shellscape removed whitespaces, is ok now?

**Comment by shellscape** [comment]
@alindesign there's another that you've missed. I commented twice on it. And you must resolve the conflicts before we can merge this.

**Comment by alindesign** [comment]
@shellscape i think this was, sorry, i prefer to use empty line before each if :p we don't have same code style :)

**Pull request merged**